src/Services/InAppPurchasesService.php

```diff
@@ -33,4 +37,27 @@ public function listPurchases(string $extensionName, Context $context): InAppPur
     {
         return $this->client->listInAppPurchases($extensionName, $context);
     }
+
+    /**
+     * @param Collection<covariant InAppPurchaseIdentifiableStruct> $allPurchases
+     * @param array<int, string> $validPurchases
+     *
+     * @return Collection<covariant InAppPurchaseIdentifiableStruct>
+     */
+    public function filterValidInAppPurchases(
+        Collection $allPurchases,
+        array $validPurchases,
+    ): Collection {
+        if (empty($validPurchases)) {
+            $allPurchases->clear();
+
+            return $allPurchases;
+        }
+
+        $allPurchases->filter(function (InAppPurchaseStruct|InAppPurchaseCartPositionStruct $purchase) use ($validPurchases) {
```

Comment:
I can't really imagine that this works. `filter` in Shopware Collections returns a new Collection and does not manipulate the old one. Therefore, in your empty case above, you manipulate the existing collection and return it, and here, you create a new one, ignore it and return the unmanipulated one.